resources/js/processes-catalogue/components/ProcessTab.vue

```diff
@@ -5,29 +5,359 @@
+          field: "due_at",
+          format: "datetime",
+          sortable: true,
+          default: true,
+          width: 140,
+        },
+      ],
+      dataRequests: {},
+      dataTasks: {},
+      savedSearch: false,
+      queryTask: "",
+      queryRequest: "",
+      perPage: 10,
+    };
+  },
+  computed: {
+    endpoint() {
+      if (this.savedSearch !== false) {
+        return `saved-searches/${this.savedSearch}/results`;
+      }
+
+      return "requests";
+    },
+  },
+  mounted() {
+    this.queryBuilder();
+  },
+  methods: {
+    jsonRows(rows) {
+      if (rows.length === 0 || !_.has(_.head(rows), "_json")) {
+        return rows;
+      }
+      return rows.map((row) => JSON.parse(row._json));
+    },
+    changePage(page) {
+      this.page = page;
+      this.queryBuilder();
+    },
+    handleRowClick(type, row) {
+      const openFunction = type === 'request' ? this.openRequest : this.openTask;
+      window.location.href = openFunction(row, 1);
+    },
+    openRequest(data, index) {
+      return `/requests/${data.id}`;
+    },
+    transform(data) {
+      // Clean up fields for meta pagination so vue table pagination can understand
+      data.meta.last_page = data.meta.total_pages;
+      data.meta.from = (data.meta.current_page - 1) * data.meta.per_page;
+      data.meta.to = data.meta.from + data.meta.count;
+      data.data = this.jsonRows(data.data);
+      for (let record of data.data) {
+        //format Status
+        record["case_number"] = this.formatCaseNumber(record);
+        record["status"] = this.formatStatus(record["status"]);
+        record["participants"] = this.formatParticipants(
+          record["participants"]
+        );
+      }
+      return data;
+    },
+    formatParticipants(participants) {
+      return {
+        component: "AvatarImage",
+        props: {
+          size: "25",
+          "input-data": participants,
+          "hide-name": false,
+        },
+      };
+    },
+    formatStatus(status) {
+      let color = "success",
+        label = "In Progress";
+      switch (status) {
+        case "DRAFT":
+          color = "danger";
+          label = "Draft";
+          break;
+        case "CANCELED":
+          color = "danger";
+          label = "Canceled";
+          break;
+        case "COMPLETED":
+          color = "primary";
+          label = "Completed";
+          break;
+        case "ERROR":
+          color = "danger";
+          label = "Error";
+          break;
+      }
+      return (
+        '<span class="badge badge-' +
+        color +
+        " status-" +
+        color +
+        '">' +
+        this.$t(label) +
+        "</span>"
+      );
+    },
+    openTask(task) {
+      return `/tasks/${task.id}/edit`;
+    },
+    formatCaseNumber(value) {
+      return `
+      <a href="${this.openRequest(value, 1)}"
+         class="text-nowrap">
+         # ${value.case_number}
+      </a>`;
+    },
+    queryBuilder() {
```

Comment:
split this method or create other method
one for requests and other for tasks

Reply:
New methods for tasks and requests were created